das/worker.go

```diff
@@ -77,14 +77,13 @@ func (w *worker) run(ctx context.Context, timeout time.Duration, resultCh chan<-
 
 	for curr := w.state.from; curr <= w.state.to; curr++ {
 		err := w.sample(ctx, timeout, curr)
-		w.setResult(curr, err)
 		if errors.Is(err, context.Canceled) {
 			// sampling worker will resume upon restart
-			break
+			return
 		}
+		w.setResult(curr, err)
```

Comment:
Why not to set err result?

Reply:
if node is shut down, last header sampling will end with context.Canceled, that will put it into failed map after restart and would be picked by a retry job. I would still sample just fine after retry tho, thats why it wasn't noticed before. I changed it to clean up stored checkpoint.

Reply:
If it is canceled, it means it didn't finish, so restart is needed or do I miss anything

Reply:
Yeah I am also confused.

Reply:
Sorry for confusion, I will try to explain by example:
`catchup` job(1-100) was canceled in the middle (lets say 50)
 
Previously: it result for 50 to be stored as failed(context.Canceled). So would be stored into checkpoint as:
```
workersToResume: [51:100]
failed: 50
 ```
 After restart it would be handled by 2 workers: `retry`(50:50) and `catchup`(51:100), so all header would be processed.
 
 Now:
 If context.Cancel happens, the worker will not be count as processed and job will be storred as :
 ```
workersToResume: [50:100]
```
After restart it will resume with one `catchup`(50:100) worker, starting from canceled height

Reply:
Okay -- super hairy tho 😭